modules/library/render/src/main/java/org/geotools/renderer/style/DynamicSymbolFactoryFinder.java

```diff
@@ -64,6 +70,26 @@ private DynamicSymbolFactoryFinder() {}
         return getServiceRegistry().getFactories(MarkFactory.class, null, null).iterator();
     }
 
+    /**
+     * Finds all implementations of {@link MarkFactory} which have registered using the services
+     * mechanism.
+     *
+     * @param hints An optional map of hints for factory configuration, or {@code null} if none.
```

Comment:
Tell the user which hints could be useful here.

Reply:
Added allowed hints.